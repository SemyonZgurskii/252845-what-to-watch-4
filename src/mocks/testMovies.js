const testMoviesData = [
  {
    title: `The Grand Budapest Hotel`,
    images: {
      card: `img/the-grand-budapest-hotel.jpg`,
      smallPoster: `img/the-grand-budapest-hotel-poster.jpg`,
      bigPoster: `img/bg-the-grand-budapest-hotel.jpg`,
    },
    genre: `Fighter`,
    releaseDate: 1994,
    rating: {score: 0.6, level: `Very good`, count: 74},
    info: {
      description:
        `Это жёсткий город - своя шкала контрастов.\n` +
        `Купеческая Мекка, государство в государстве.\n` +
        `Тот самый город, что не проводит гей-парады.\n` +
        `Тот самый город - профицит которого ярды.\n` +
        `\n` +
        `Плитка сыпется с небес, тебе нравится.\n` +
        `Эти штрафы и от них не избавиться.\n` +
        `Патриаршие и пруды, центр варится.\n` +
        `Рейды на Садовом. Ночь. Снова пятница.\n` +
        `\n` +
        `В пятом поколении коренной москвич.\n` +
        `Не хожу на митинги и не втираю дичь.\n` +
        `Помню этот город пятнадцать лет назад -\n` +
        `Отвечаю, он щас мировой стандарт.\n` +
        `\n` +
        `Квинтэссенция интеллигентов и джамшутов.\n` +
        `Здесь комерсы выходят в окна без парашюта.\n` +
        `Пешком с Арбата и до площади Гагарина -\n` +
        `Там хлопну бургер за здоровье Собянина.`,
      director: `Timati`,
      stars: [`Timati`, `Goof`],
    },
    preview: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
  },
  {
    title: `Johnny English`,
    images: {
      card: `img/johnny-english.jpg`,
      smallPoster: `img/johnny-english-poster.jpg`,
      bigPoster: `img/bg-johnny-english.jpg`,
    },
    genre: `Fighter`,
    releaseDate: 1992,
    rating: {score: 2.9, level: `Bad`, count: 68},
    info: {
      description:
        `Это жёсткий город - своя шкала контрастов.\n` +
        `Купеческая Мекка, государство в государстве.\n` +
        `Тот самый город, что не проводит гей-парады.\n` +
        `Тот самый город - профицит которого ярды.\n` +
        `\n` +
        `Плитка сыпется с небес, тебе нравится.\n` +
        `Эти штрафы и от них не избавиться.\n` +
        `Патриаршие и пруды, центр варится.\n` +
        `Рейды на Садовом. Ночь. Снова пятница.\n` +
        `\n` +
        `В пятом поколении коренной москвич.\n` +
        `Не хожу на митинги и не втираю дичь.\n` +
        `Помню этот город пятнадцать лет назад -\n` +
        `Отвечаю, он щас мировой стандарт.\n` +
        `\n` +
        `Квинтэссенция интеллигентов и джамшутов.\n` +
        `Здесь комерсы выходят в окна без парашюта.\n` +
        `Пешком с Арбата и до площади Гагарина -\n` +
        `Там хлопну бургер за здоровье Собянина.`,
      director: `Timati`,
      stars: [`Timati`, `Goof`],
    },
    preview: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
  },
  {
    title: `Fantastic Beasts: The Crimes of Grindelwald`,
    images: {
      card: `img/fantastic-beasts-the-crimes-of-grindelwald.jpg`,
      smallPoster: `img/fantastic-beasts-the-crimes-of-grindelwald-poster.jpg`,
      bigPoster: `img/bg-fantastic-beasts-the-crimes-of-grindelwald.jpg`,
    },
    genre: `Drama`,
    releaseDate: 2001,
    rating: {score: 5.1, level: `Awesome`, count: 16},
    info: {
      description:
        `Это жёсткий город - своя шкала контрастов.\n` +
        `Купеческая Мекка, государство в государстве.\n` +
        `Тот самый город, что не проводит гей-парады.\n` +
        `Тот самый город - профицит которого ярды.\n` +
        `\n` +
        `Плитка сыпется с небес, тебе нравится.\n` +
        `Эти штрафы и от них не избавиться.\n` +
        `Патриаршие и пруды, центр варится.\n` +
        `Рейды на Садовом. Ночь. Снова пятница.\n` +
        `\n` +
        `В пятом поколении коренной москвич.\n` +
        `Не хожу на митинги и не втираю дичь.\n` +
        `Помню этот город пятнадцать лет назад -\n` +
        `Отвечаю, он щас мировой стандарт.\n` +
        `\n` +
        `Квинтэссенция интеллигентов и джамшутов.\n` +
        `Здесь комерсы выходят в окна без парашюта.\n` +
        `Пешком с Арбата и до площади Гагарина -\n` +
        `Там хлопну бургер за здоровье Собянина.`,
      director: `Timati`,
      stars: [`Timati`, `Goof`],
    },
    preview: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
  },
  {
    title: `Bohemian Rhapsody`,
    images: {
      card: `img/bohemian-rhapsody.jpg`,
      smallPoster: `img/bohemian-rhapsody-poster.jpg`,
      bigPoster: `img/bg-bohemian-rhapsody.jpg`,
    },
    genre: `Fighter`,
    releaseDate: 2004,
    rating: {score: 2, level: `Normal`, count: 28},
    info: {
      description:
        `Это жёсткий город - своя шкала контрастов.\n` +
        `Купеческая Мекка, государство в государстве.\n` +
        `Тот самый город, что не проводит гей-парады.\n` +
        `Тот самый город - профицит которого ярды.\n` +
        `\n` +
        `Плитка сыпется с небес, тебе нравится.\n` +
        `Эти штрафы и от них не избавиться.\n` +
        `Патриаршие и пруды, центр варится.\n` +
        `Рейды на Садовом. Ночь. Снова пятница.\n` +
        `\n` +
        `В пятом поколении коренной москвич.\n` +
        `Не хожу на митинги и не втираю дичь.\n` +
        `Помню этот город пятнадцать лет назад -\n` +
        `Отвечаю, он щас мировой стандарт.\n` +
        `\n` +
        `Квинтэссенция интеллигентов и джамшутов.\n` +
        `Здесь комерсы выходят в окна без парашюта.\n` +
        `Пешком с Арбата и до площади Гагарина -\n` +
        `Там хлопну бургер за здоровье Собянина.`,
      director: `Timati`,
      stars: [`Timati`, `Goof`],
    },
    preview: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
  },
  {
    title: `Macbeth`,
    images: {
      card: `img/macbeth.jpg`,
      smallPoster: `img/macbeth-poster.jpg`,
      bigPoster: `img/bg-macbeth.jpg`,
    },
    genre: `Fighter`,
    releaseDate: 2011,
    rating: {score: 7.6, level: `Bad`, count: 60},
    info: {
      description:
        `Это жёсткий город - своя шкала контрастов.\n` +
        `Купеческая Мекка, государство в государстве.\n` +
        `Тот самый город, что не проводит гей-парады.\n` +
        `Тот самый город - профицит которого ярды.\n` +
        `\n` +
        `Плитка сыпется с небес, тебе нравится.\n` +
        `Эти штрафы и от них не избавиться.\n` +
        `Патриаршие и пруды, центр варится.\n` +
        `Рейды на Садовом. Ночь. Снова пятница.\n` +
        `\n` +
        `В пятом поколении коренной москвич.\n` +
        `Не хожу на митинги и не втираю дичь.\n` +
        `Помню этот город пятнадцать лет назад -\n` +
        `Отвечаю, он щас мировой стандарт.\n` +
        `\n` +
        `Квинтэссенция интеллигентов и джамшутов.\n` +
        `Здесь комерсы выходят в окна без парашюта.\n` +
        `Пешком с Арбата и до площади Гагарина -\n` +
        `Там хлопну бургер за здоровье Собянина.`,
      director: `Timati`,
      stars: [`Timati`, `Goof`],
    },
    preview: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
  },
  {
    title: `Aviator`,
    images: {
      card: `img/aviator.jpg`,
      smallPoster: `img/aviator-poster.jpg`,
      bigPoster: `img/bg-aviator.jpg`,
    },
    genre: `Drama`,
    releaseDate: 2002,
    rating: {score: 2.2, level: `Very good`, count: 22},
    info: {
      description:
        `Это жёсткий город - своя шкала контрастов.\n` +
        `Купеческая Мекка, государство в государстве.\n` +
        `Тот самый город, что не проводит гей-парады.\n` +
        `Тот самый город - профицит которого ярды.\n` +
        `\n` +
        `Плитка сыпется с небес, тебе нравится.\n` +
        `Эти штрафы и от них не избавиться.\n` +
        `Патриаршие и пруды, центр варится.\n` +
        `Рейды на Садовом. Ночь. Снова пятница.\n` +
        `\n` +
        `В пятом поколении коренной москвич.\n` +
        `Не хожу на митинги и не втираю дичь.\n` +
        `Помню этот город пятнадцать лет назад -\n` +
        `Отвечаю, он щас мировой стандарт.\n` +
        `\n` +
        `Квинтэссенция интеллигентов и джамшутов.\n` +
        `Здесь комерсы выходят в окна без парашюта.\n` +
        `Пешком с Арбата и до площади Гагарина -\n` +
        `Там хлопну бургер за здоровье Собянина.`,
      director: `Timati`,
      stars: [`Timati`, `Goof`],
    },
    preview: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
  },
  {
    title: `We need to talk about Kevin`,
    images: {
      card: `img/we-need-to-talk-about-kevin.jpg`,
      smallPoster: `img/we-need-to-talk-about-kevin-poster.jpg`,
      bigPoster: `img/bg-we-need-to-talk-about-kevin.jpg`,
    },
    genre: `Drama`,
    releaseDate: 2007,
    rating: {score: 0.3, level: `Very good`, count: 7},
    info: {
      description:
        `Это жёсткий город - своя шкала контрастов.\n` +
        `Купеческая Мекка, государство в государстве.\n` +
        `Тот самый город, что не проводит гей-парады.\n` +
        `Тот самый город - профицит которого ярды.\n` +
        `\n` +
        `Плитка сыпется с небес, тебе нравится.\n` +
        `Эти штрафы и от них не избавиться.\n` +
        `Патриаршие и пруды, центр варится.\n` +
        `Рейды на Садовом. Ночь. Снова пятница.\n` +
        `\n` +
        `В пятом поколении коренной москвич.\n` +
        `Не хожу на митинги и не втираю дичь.\n` +
        `Помню этот город пятнадцать лет назад -\n` +
        `Отвечаю, он щас мировой стандарт.\n` +
        `\n` +
        `Квинтэссенция интеллигентов и джамшутов.\n` +
        `Здесь комерсы выходят в окна без парашюта.\n` +
        `Пешком с Арбата и до площади Гагарина -\n` +
        `Там хлопну бургер за здоровье Собянина.`,
      director: `Timati`,
      stars: [`Timati`, `Goof`],
    },
    preview: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
  },
  {
    title: `Revenant`,
    images: {
      card: `img/revenant.jpg`,
      smallPoster: `img/revenant-poster.jpg`,
      bigPoster: `img/bg-revenant.jpg`,
    },
    genre: `Horror`,
    releaseDate: 2011,
    rating: {score: 0.3, level: `Awesome`, count: 14},
    info: {
      description:
        `Это жёсткий город - своя шкала контрастов.\n` +
        `Купеческая Мекка, государство в государстве.\n` +
        `Тот самый город, что не проводит гей-парады.\n` +
        `Тот самый город - профицит которого ярды.\n` +
        `\n` +
        `Плитка сыпется с небес, тебе нравится.\n` +
        `Эти штрафы и от них не избавиться.\n` +
        `Патриаршие и пруды, центр варится.\n` +
        `Рейды на Садовом. Ночь. Снова пятница.\n` +
        `\n` +
        `В пятом поколении коренной москвич.\n` +
        `Не хожу на митинги и не втираю дичь.\n` +
        `Помню этот город пятнадцать лет назад -\n` +
        `Отвечаю, он щас мировой стандарт.\n` +
        `\n` +
        `Квинтэссенция интеллигентов и джамшутов.\n` +
        `Здесь комерсы выходят в окна без парашюта.\n` +
        `Пешком с Арбата и до площади Гагарина -\n` +
        `Там хлопну бургер за здоровье Собянина.`,
      director: `Timati`,
      stars: [`Timati`, `Goof`],
    },
    preview: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
  },
];

export {testMoviesData};
