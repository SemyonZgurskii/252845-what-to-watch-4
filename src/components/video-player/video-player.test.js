import React from 'react';
import renderer from 'react-test-renderer';
import VideoPlayer from './video-player.jsx';
import {testMoviesData} from '../../mocks/testMovies.js';

describe(`Create VideoPlayer snapshot`, () => {
  it(`Should correctly rendeo VideoPlayer`, () => {
    const isPlaying = true;
    const {preview, images, title} = testMoviesData[0];

    const VideoPlayerElement = renderer.create(
        <VideoPlayer
          previewImage = {images.card}
          previewVideo = {preview}
          title = {title}
          isPlaying = {isPlaying}
        />
    ).toJSON();

    expect(VideoPlayerElement).toMatchSnapshot();
  });
});
