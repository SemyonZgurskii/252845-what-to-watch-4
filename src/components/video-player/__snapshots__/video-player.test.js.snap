// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Create VideoPlayer snapshot Should correctly rendeo VideoPlayer 1`] = `
<video
  aria-label="The Grand Budapest Hotel movie preview"
  height="175"
  muted={true}
  poster="img/the-grand-budapest-hotel.jpg"
  preload="none"
  width="280"
>
  <source
    src="https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4"
    type="video/mp4"
  />
</video>
`;
